lib/src/main/java/at/bitfire/cert4android/UserDecisionRegistry.kt

```diff
@@ -94,11 +94,8 @@ class UserDecisionRegistry private constructor(
 
         // continue work that's waiting for decisions
         synchronized(pendingDecisions) {
-            pendingDecisions[cert]?.iterator()?.let { iter ->
-                while (iter.hasNext()) {
-                    iter.next().resume(trusted)
-                    iter.remove()
```

Comment:
As far as I see, this change doesn't call `remove()`.

So it wouldn't remove the finished entries from the list?

I think the iterator way is needed because a loop + explicitly removing entries of the iterate list may not be well-defined and cause problems.

Reply:
ohh, right. I was a bit too quick then. I thought the `.iterator()` call would create it's own internal list of Continuation elements. Then it would not be any differen from a normal loop. But I can see the kdoc of `.remove()` clearly says that it "Removes from the underlying collection the last element returned by this iterator." So it removes the elements from the actual `pendingDecisions` map. :+1: